foreman.te

```diff
@@ -518,3 +518,15 @@ allow httpd_t foreman_lib_t:lnk_file { getattr read };
 # and manage links
 allow foreman_rails_t tmp_t:file map;
 allow foreman_rails_t tmp_t:lnk_file { create unlink };
+
+######################################
+#
+# Foreman Bootdisk plugin
+#
+
+# The plugin spawns genisoimage which needs access to /dev/shm
+require {
+  type fs_t;
+}
+allow foreman_rails_t tmpfs_t:filesystem getattr;
+allow foreman_rails_t fs_t:filesystem getattr;
```

Comment:
I can see `/dev/shm` itself is `tmpfs_t`, but why `fs_t`? I don't even see it on my Fedora system when I use `semanage fcontext -l | grep :fs_t`.

Reply:
The commit message is not exactly correct (anymore).
I didn't dig super deep, but the `tmpfs_t` one is sufficient to create a bootdisk from the menu.
But when you try to provision using the bootdisk method, you get:
```
time->Mon Jun  3 05:05:39 2024
type=PROCTITLE msg=audit(1717405539.843:38323): proctitle=67656E69736F696D616765002D6F002F746D702F626F6F746469736B32303234303630332D3238313535312D6176316F34322E69736F002D69736F2D6C6576656C0032002D620069736F6C696E75782E62696E002D6300626F6F742E636174002D6E6F2D656D756C2D626F6F74002D626F6F742D6C6F61642D73697A650034
type=SYSCALL msg=audit(1717405539.843:38323): arch=c000003e syscall=137 success=yes exit=0 a0=5583c49027c0 a1=7ffc3f861390 a2=28 a3=0 items=0 ppid=318892 pid=318893 auid=4294967295 uid=986 gid=987 euid=986 suid=986 fsuid=986 egid=987 sgid=987 fsgid=987 tty=(none) ses=4294967295 comm="genisoimage" exe="/usr/bin/xorriso" subj=system_u:system_r:foreman_rails_t:s0 key=(null)
type=AVC msg=audit(1717405539.843:38323): avc:  denied  { getattr } for  pid=318893 comm="genisoimage" name="/" dev="vda4" ino=128 scontext=system_u:system_r:foreman_rails_t:s0 tcontext=system_u:object_r:fs_t:s0 tclass=filesystem permissive=0
```
and boom, nothing works.

Reply:
I was surprised by it, but I can find it in the source: https://github.com/fedora-selinux/selinux-policy/blob/eb5635fb3c3b1b59b3643910ca16360dc1c5e8c3/policy/modules/kernel/filesystem.te#L14-L17